chameleon/tests/integration_tests.rs

```diff
@@ -0,0 +1,58 @@
+use bitcoin::Network;
+use chameleon::ExchangeRatesProvider;
+use graphql::perro::Error;
+use honey_badger::secrets::{derive_keys, generate_keypair, generate_mnemonic};
+use honey_badger::{Auth, AuthLevel};
+use simplelog::TestLogger;
+use std::env;
+use std::sync::{Arc, Once};
+
+static INIT_LOGGER_ONCE: Once = Once::new();
+
+#[cfg(test)]
+#[ctor::ctor]
+fn init() {
+    INIT_LOGGER_ONCE.call_once(|| {
+        TestLogger::init(simplelog::LevelFilter::Info, simplelog::Config::default()).unwrap();
+    });
+}
+
+#[test]
+fn test_list_currency_codes() {
+    let provider = build_provider();
+    let list = provider.list_currency_codes().unwrap();
+    assert_eq!(list, vec!["CHF", "EUR", "SATS", "USD"]);
```

Comment:
Should we test this in this way? If the currency code list wasn't expected to change, we wouldn't even need to query it. What about we just check that it contains some expected currency codes (e.g. EUR, USD and CHF) but allow others to be added?